great_expectations/core/validation_config.py

```diff
@@ -215,14 +215,15 @@ def run(
             validation_result_id,
         ) = self._get_expectation_suite_and_validation_result_ids(validator)
 
-        store_validation_results(
-            validation_result_store=self._validation_results_store,
+        ref = self._validation_results_store.store_validation_results(
             suite_validation_result=results,
             suite_validation_result_identifier=validation_result_id,
             expectation_suite_identifier=expectation_suite_identifier,
-            cloud_mode=self._validation_results_store.cloud_mode,
         )
 
+        if isinstance(ref, GXCloudResourceRef):
+            results.result_url = self._validation_results_store.get_result_url(ref)
+
```

Comment:
All of these can probably be brought together but that can be done after the store validation result action is removed